pkg/alertmanager/statefulset.go

```diff
@@ -66,6 +66,7 @@ func makeStatefulSet(am *monitoringv1.Alertmanager, config Config, inputHash str
 	if am.Spec.Replicas != nil && *am.Spec.Replicas < 0 {
 		am.Spec.Replicas = &intZero
 	}
+	// TODO(slashpai): Remove this assignment after v0.57 since this is handled at CRD level
```

Comment:
I don't think we can remove this so soon as there is no guarantee that users will update CRDs when updating operator. Let's target at least 0.60 release for the removal of this part of code.